src/FakeItEasy/Configuration/ArgumentCollection.cs

```diff
@@ -68,6 +69,8 @@ public int Count
         /// </summary>
         public IEnumerable<string> ArgumentNames { get; }
 
+        internal MethodInfo Method { get; }

```

Comment:
Here's an idea **that I have not spiked**. `Matches(IFakeObjectCall)` is our sledgehammer, essentially. It can do anything any of the other calls do.
Our internal code could always use this. Everything internal gets passed an `IFakeObjectCall` and we augment the predicates to deal with them.

Alternatively, we could at least drop the other constructor for `ArgumentCollection`.